engine/src/tangl/story/fabula/address_resolver.py

```
@@ -198,9 +387,48 @@ def ensure_instance(
+        if obj_cls is Subgraph or (isinstance(obj_cls, type) and issubclass(obj_cls, Subgraph)):
+            logger.debug(
+                "Reusing existing subgraph at '%s' (template expects container type)",
+                address,
+            )
+            return existing_subgraph
+        obj_cls_name = obj_cls if isinstance(obj_cls, str) else obj_cls.__name__
+        raise TypeError(
+            f"Cannot create instance at '{address}': "
+            f"Subgraph already exists but template expects {obj_cls_name}. "
+            "This indicates a namespace container was incorrectly created at an instance address."
+        )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)